src/tailwindcss-dotnet/Cli/Upstream.cs

```diff
@@ -1,21 +1,42 @@
 ﻿using System.Runtime.InteropServices;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
 
 namespace Tailwindcss.DotNetTool.Cli;
 
 public class Upstream
 {
-    public static string Version => "v4.0.0";
+    private static readonly HttpClient httpClient = new HttpClient();
+
+    public static async Task<string> GetLatestReleaseVersionAsync()
+    {
+        var fallbackVersion = "v4.0.0";
+        try
+        {
+            httpClient.DefaultRequestHeaders.UserAgent.ParseAdd("dotnet-tailwindcss");
+            var response = await httpClient.GetAsync("https://api.github.com/repos/tailwindlabs/tailwindcss/releases/latest");
+            response.EnsureSuccessStatusCode();
+
+            var json = await response.Content.ReadAsStringAsync();
+            using var document = JsonDocument.Parse(json);
+            var version = document.RootElement.GetProperty("tag_name").GetString();
+
+            return version ?? fallbackVersion;
+        }
+       
+        catch
+        {
+            return fallbackVersion;
+        }
+    }
+    public static string Version{ get; } = GetLatestReleaseVersionAsync().GetAwaiter().GetResult();
```

Comment:
This property is called twice in a blocking manner, which I would like to avoid. It can be converted into a method, and the result can be cached to prevent repeated calls to the remote address.